renku/core/management/datasets.py

```diff
@@ -1296,14 +1295,14 @@ def checkout(repo, ref):
 
         depth = 1 if not ref else None
         ref = ref or renku_branch
-        u = GitURL.parse(url)
-        path = u.pathname
-        if u.hostname == "localhost":
+        u = validate_url(url)
+        path = u.path
+        if "localhost" in u.netloc:
```

Comment:
Shouldn't it be `u.netloc = "localhost"` ? In case of myawesomelocalhost.com (Domain still available if someone wants it!)

Reply:
yeah that was my first thought, but the idea behind this is that to support `localhost:<PORT>`. but indeed it fails big time, so i'll replace it with a regex. i guess `127.*` should be here as for check, or?

Reply:
Honestly I'm not sure what case this part covers. it's not the `file://path/to/.git` case. But I don't see how for `git://localhost/myfolder` we can just assume that `str(Path(path).resolve())` actually resolves to anything on disk?

Reply:
true that!